### PR TITLE
support a certificate contains both client and server auth

### PR DESCRIPTION
**Pull request by chenzhiwei**
Some applications need Key Usage `TLS Web Server Authentication` and `TLS Web Client Authentication` in a single certificate.

This PR is used to fill this gap, which allows easyrsa to generate server-client certificate.

### CODE REVIEW

File: easyrsa3/easyrsa [hunk 1]
```diff
@@ -689,14 +691,15 @@ $(display_dn req "$req_in")
...
+				server-client)	print "nsCertType = server" ;;
 				client)	print "nsCertType = client" ;;
 				ca)	print "nsCertType = sslCA" ;;
 			esac
 		fi
 
 		# If type is server and no subjectAltName was requested,
 		# add one to the extensions file
-		if [ "$crt_type" = 'server' ];
+		if [ "$crt_type" = 'server' ] or [ "$crt_type" = 'server-client' ];
```
Comment (ecrist) [inline easyrsa3/easyrsa]: The syntax here is odd.  This should be:

`if [ "$crt_Type" = "server" -o "$crt_type" = "server-client" ];`

File: easyrsa3/easyrsa [hunk 2]
```diff
@@ -88,10 +89,11 @@ cmd_help() {
 
       This request file must exist in the reqs/ dir and have a .req file
       extension. See import-req below for importing reqs from other sources." ;;
-		build|build-client-full|build-server-full) text="
+		build|build-client-full|build-server-full|build-server-client-full) text="
   build-client-full <filename_base> [ cmd-opts ]
   build-server-full <filename_base> [ cmd-opts ]
-      Generate a keypair and sign locally for a client or server
+  build-server-client-full <filename_base> [ cmd-opts ]
+      Generate a keypair and sign locally for a client or server or both
```
Comment (ecrist) [inline easyrsa3/easyrsa]: The help text here could be a bit more verbose, explaining that this option provides both server and client certificate extensions.  This makes it sound like it's still not decided.

File: easyrsa3/x509-types/server-client [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# X509 extensions for a server
```
Comment (ecrist) [inline easyrsa3/x509-types/server-client]: This should illustrate both server and client.

**Comment by chenzhiwei** [comment]
@ecrist Thank you for reviewing my PR, updated the PR according to you comments.

**Comment by TinCanTech** [comment]
Support for `serverClient` certificate was added:
* https://github.com/OpenVPN/easy-rsa/commit/bebd71e76fcc0e01014cc67194e41db8fe16fdad

This PR should probably be closed.

**Comment by chenzhiwei** [comment]
thank you!

**Pull request closed without merging**